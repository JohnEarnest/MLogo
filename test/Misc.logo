# Misc
#
# test miscellaneous functionality

words

local 'QUUX [1 2 3]

words

erase 'QUUX

words

# I had an amusing bug in which procedure bodies
# containing only a call to one primitive were
# confused with the primitive body itself by
# the TCO mechanism, leading to infinite loops:

to foo
	trace
end

to bar
	trace
	run [foo]
end
bar