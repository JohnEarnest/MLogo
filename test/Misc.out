args ascall asname asvalue bind butfirst butlast difference equal? erase first flatten fput greater? if item join last less? list? local lput make member num? output print println product quotient random readlist remainder repeat run size stop sum thing trace unless version word? words 

'QUUX args ascall asname asvalue bind butfirst butlast difference equal? erase first flatten fput greater? if item join last less? list? local lput make member num? output print println product quotient random readlist remainder repeat run size stop sum thing trace unless version word? words 

args ascall asname asvalue bind butfirst butlast difference equal? erase first flatten fput greater? if item join last less? list? local lput make member num? output print println product quotient random readlist remainder repeat run size stop sum thing trace unless version word? words 

trace: 
	'bar

trace: 
	'foo
	'bar

